source/checksums.c

```diff
@@ -14,6 +20,7 @@ void aws_checksums_library_init(struct aws_allocator *allocator) {
 
         aws_common_library_init(allocator);
 
+        (void)aws_cpu_has_clmul_cached(); /* warm up the cache */
```

Comment:
trivial: maybe just rename s_init_detection_cache() -> `aws_checksums_init_detection_cache()` and call that directly from there. That's makes it a bit more obvious than calling a function that has the side effect of calling the function you want.

I understand the `/* warm up the cache*/` comment, but it might not make as much sense to some future coder looking at this line.